File: solutions/2275. Largest Combination With Bitwise AND Greater Than Zero/2275.cpp
// Time complexity: O(n)
// Space complexity: O(1)
class Solution {
 public:
  int largestCombination(const vector<int>& candidates) {
    constexpr int kMaxBit = 24;
    int ans = 0;

    for (int i = 0; i < kMaxBit; ++i) {
      int count = 0;
      for (const int candidate : candidates)
        if (candidate >> i & 1)
          ++count;
      ans = max(ans, count);
    }

    return ans;
  }
};

File: solutions/2275. Largest Combination With Bitwise AND Greater Than Zero/2275.rs
// Time complexity: O(n)
// Space complexity: O(1)
impl Solution {
    pub fn largest_combination(candidates: Vec<i32>) -> i32 {
        (0..24)
            .map(|i| candidates.iter().filter(|&&c| (c >> i) & 1 == 1).count() as i32)
            .max()
            .unwrap_or(0)
    }
}
